### DISCUSSION TITLE
How can I add thread::sleep to the async event loop

### DISCUSSION THREAD

**Discussion by mjehrhart** [bytebeamio/rumqtt#876] [Q&A]
I have an app which runs async rumqtt. In the app, the user can delay receiving messages.  To do this, I thought I could add thread::sleep(120) to the code to delay the messages. I tried using tokio::time::sleep as well.  But this keeps closing the connection with this error message: "CERROR::ConnectionError::Mqtt state: Io error: Custom { kind: ConnectionAborted, error: "connection closed by peer" }".

Any suggestions to make this work? or is it not possible?

**Comment by swanandx** [comment]
what is the keep alive interval you are connecting with? If it's less rhat 120, and you are not polling the eventloop, that means there won't be any Ping requests and hence broker might close the connection.

**Reply by mjehrhart** [reply]
Ok, I didn’t know that. The delay would be  from 30 minutes to 4 hours. But I was testing with 10 seconds.  I’ll try out 120 seconds then. 

Thanks for this.

**Reply by swanandx** [reply]
I said 120 because of in original message you mentioned `thread::sleep(120)`, and hence i thought for 120 seconds you won't poll the eventloop.


I'm curious what's the use of delay of receiving by 30 mins - 4 hrs? Can't we just send it after that interval?

**Comment by mjehrhart** [comment]
So I tried a variety of values and none work. If I call sleep during the event loop, the connection is broken. 

I have an app that listens to mqtt for certain messages and then send an iOS push notification to the user. There can be a lot of notifications so some people want to “snooze” notifications for x amount of time.
